arm_isa.cpp

```diff
@@ -1377,7 +1377,7 @@ inline void LDRH(arm_isa* ref, int rd, int rn,
     printf("Unpredictable LDRH instruction result (Address is not Halfword Aligned)\n");
     return;
   }
-  value = MEM.read(ref->ls_address.entire);
+  value = MEM.read_half(ref->ls_address.entire);
   value &= 0xFFFF; /* Zero extends halfword value 
 		      BUG: Model must be little endian in order to the code work  */
```

Comment:
I guess we can now remove this line "BUG: Model must be little endian". I believe this was written when "read" was being used to read 4 bytes and this mask was used to get half word. It is true it should work on little endian hosts.